src/components/Profile/StakingTokens.tsx

```diff
@@ -0,0 +1,229 @@
+import { useHistory } from 'react-router-dom'
+import {
+  Box,
+  Distribution,
+  GU,
+  shortenAddress,
+  useTheme,
+  useViewport,
+} from '@1hive/1hive-ui'
+
+import BigNumber from '@lib/bigNumber'
+import { useWallet } from '@providers/Wallet'
+import { stakesPercentages } from '@utils/math-utils'
+import { getNetworkType } from '@/utils/web3-utils'
+import { useGardens } from '@/providers/Gardens'
+
+const DISTRIBUTION_ITEMS_MAX = 6
+
+type StakeItem = {
+  amount?: any
+  gardenId: string | null
+  proposalId: string | null
+  proposalName: string
+}
+
+type TransformedStakeType = {
+  item: StakeItem
+  percentage: any
+} | null
+
+type StakingTokensProps = {
+  myStakes: Array<StakeItem>
+}
+
+function displayedStakes(stakes: Array<StakeItem>, total: BigNumber) {
+  return stakesPercentages(
+    stakes.map(({ amount }) => amount),
+    {
+      total,
+      maxIncluded: DISTRIBUTION_ITEMS_MAX,
+    }
+  ).map((stake: any) => ({
+    item: {
+      gardenId: stake.index === -1 ? null : stakes[stake.index].gardenId,
+      proposalId: stake.index === -1 ? null : stakes[stake.index].proposalId,
+      proposalName:
+        stake.index === -1 ? 'Others' : stakes[stake.index].proposalName,
+    },
+    percentage: stake.percentage,
+  }))
+}
+
+const getStakeItemsPerGarden = (
+  stakes: Array<TransformedStakeType> | null,
+  garden: string
+) => {
+  const items: Array<any> = []
+
+  stakes?.map((stake) => {
+    if (stake?.item.gardenId === garden) {
+      items.push({
+        item: stake?.item,
+        percentage: Math.round(stake?.percentage) ?? '0',
+      })
+    }
+  })
+
+  return items
+}
+
+type Garden = { address: string; name: string }
+
+const getGardenNameByGardens = (
+  gardens: Array<Garden>,
+  gardenAddress: string
+) => {
+  const garden = gardens.find(
+    (garden: Garden) => garden.address.toLowerCase() === gardenAddress
+  )
+
+  return garden?.name ?? ''
+}
+
+const getStakesByGarden = (
+  gardensMetadata: Array<Garden>,
+  stakes: Array<TransformedStakeType> | null
+): Array<{
+  garden: string
+  gardenName: string
+  items: Array<TransformedStakeType>
+}> => {
+  if (stakes === null) return []
+
+  const gardens: Array<any> = []
+  const stakesPerGarden: Array<any> = []
+
+  // get all unique gardens
+  stakes.map((stake: TransformedStakeType) => {
+    const gardenId = stake?.item?.gardenId
+    if (!gardens.includes(gardenId) && gardenId !== null) {
+      gardens.push(gardenId)
+    }
+  })
+
+  gardens.map((garden) => {
+    stakesPerGarden.push({
+      garden,
+      gardenName: getGardenNameByGardens(gardensMetadata, garden),
+      items: getStakeItemsPerGarden(stakes, garden),
+    })
+  })
+
+  return stakesPerGarden
+}
+
+function StakingTokens({ myStakes }: StakingTokensProps) {
+  const { gardensMetadata } = useGardens()
+  const theme = useTheme()
+  const { below } = useViewport()
+  const compact = below('large')
+
+  const myActiveTokens = useMemo(() => {
```

Comment:
We don't really need `useMemo` here because if the data of the `myStakes` changes, then the `useMemo` is of no use.

And moreover, it is not such heavy computation, so we are just over-complicating things here.

Reply:
It was already there. I will leave it be.